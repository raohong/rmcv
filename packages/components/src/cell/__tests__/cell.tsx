import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Cell, cellClassNames } from '..';

const testId = 'cell';

it('render correctly', () => {
  const tree = render(<Cell />);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with title', () => {
  render(<Cell title='title' />);

  expect(screen.getByText('title')).not.toBeNull();
});

it('render with value', () => {
  render(<Cell value='value' />);

  expect(screen.getByText('value')).not.toBeNull();
});

it('render with label', () => {
  render(<Cell label='label' />);

  expect(screen.getByText('label')).not.toBeNull();
});

it('render with icon', () => {
  render(<Cell icon={<span>icon</span>} />);

  expect(screen.getByText('icon')).toBeInTheDocument();
});

it('render with rightIcon', () => {
  render(<Cell rightIcon={<span data-testid='right-icon'>rightIcon</span>} />);

  expect(screen.getByTestId('right-icon')).toBeInTheDocument();
});

it('render with clickable', () => {
  render(<Cell clickable data-testid={testId} />);

  expect(screen.getByTestId(testId)).toHaveStyleRule('cursor', 'pointer');
});

it('render with isLink', () => {
  const { container } = render(<Cell isLink data-testid={testId} />);

  expect(screen.getByTestId(testId)).toHaveStyleRule('cursor', 'pointer');
  expect(
    container.querySelector(`.${cellClassNames.rightIcon}`),
  ).toBeInTheDocument();
});

it('render with center', () => {
  render(<Cell center data-testid={testId} />);

  expect(screen.getByTestId(testId)).toHaveStyleRule('align-items', 'center');
});

it('render with arrowDirection', () => {
  render(<Cell isLink arrowDirection='down' data-testid={testId} />);

  expect(screen.getByLabelText('Arrow-Down')).toBeInTheDocument();
});

it('render with onClick', async () => {
  const handler = jest.fn();

  render(<Cell onClick={handler} data-testid={testId} />);

  await userEvent.click(screen.getByTestId(testId));
  expect(handler).toHaveBeenCalled();
});

it('render with children', () => {
  render(<Cell>child</Cell>);

  expect(screen.getByText('child')).toHaveClass(cellClassNames.value);
});

it('render with size', () => {
  render(<Cell data-testid={testId} size='large' />);

  expect(screen.getByTestId(testId)).toHaveClass(cellClassNames.sizeLarge);
});
