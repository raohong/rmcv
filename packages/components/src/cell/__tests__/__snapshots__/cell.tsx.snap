// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: inherit;
  border: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
  cursor: pointer;
  line-height: 24px;
  padding: 10px 16px;
  font-size: 14px;
  color: #323233;
  background: #fff;
  border-color: #ebedf0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: auto;
}

.emotion-0:not(:last-child)::after {
  content: "";
  position: absolute;
  left: -50%;
  top: -50%;
  bottom: -50%;
  right: -50%;
  border-style: solid;
  border-width: 0;
  -webkit-transform: scale(0.5);
  -moz-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  pointer-events: none;
  border-color: #ebedf0;
  border-bottom-width: 1px;
}

.emotion-0:not(:last-child)::after {
  margin-left: calc(2 * 16px);
  margin-right: calc(2 * 16px);
}

<div
    class="RVuiTouchable-root RVuiCell-root RVuiCell-sizeNormal emotion-0"
    role="button"
    type="button"
  />
</DocumentFragment>
`;
